### DISCUSSION TITLE
How to perform a Keywords search (123TEXTQUERY321)

### DISCUSSION THREAD

**Discussion by ilmazzelli** [Snow-Shell/servicenow-powershell#257] [Q&A]
Is it possible to perform a keyword search and, if so, what's the correct syntax to do that?
Thanks in advance

**Comment by gdbarron** [comment]
Hi @ilmazzelli.  Are you wanting to search across the entire instance, specific table/field?

**Reply by ilmazzelli** [reply]
Hi @gdbarron, 
I wanna search across all fields of a specific table (incident) a particular word.
For example search for "komodosvrgrp" across all fields of table incident
We just perform this search using 123TEXTQUERY321=komodosvrgrp in sysparm_query calling Table API
Thanks

**Reply by gdbarron** [reply]
I'm not sure of any way to do this with the api natively.  The closest thing I can think of is to get all the properties for 1 record, determine which are strings and build a query that checks those fields for the value.

**Reply by ilmazzelli** [reply]
OK, thank you @gdbarron 
I tried following call, and it seems to me that it works as I expected.

$FilterKeyword =   
            @('sys_updated_on', '-ge', (Get-Date).AddHours(-12)),
            'and',
            @('123TEXTQUERY321', '-eq', 'komdosvrgrp')

Get-ServiceNowRecord -Table incident -Filter $FilterKeyword -Property 'number','state','opened_at','closed_at','resolved_at','short_description','u_business_service','u_application_ci'

Just another question... why for u_business_service and u_application_ci fields I get display_value and link? Is it possible to get display value only? I tried to add -DisplayValue 'true' parameter but it doesn't works

Thanks
